Source/Core/DolphinWX/Config/GameCubeConfigPane.cpp

```diff
@@ -267,12 +270,30 @@ void GameCubeConfigPane::OnSP1Changed(wxCommandEvent& event)
 
 void GameCubeConfigPane::OnSlotAButtonClick(wxCommandEvent& event)
 {
-  ChooseSlotPath(true, SConfig::GetInstance().m_EXIDevice[0]);
+  if (!m_exi_devices[0]->GetStringSelection().compare(_(EXIDEV_MIC_STR)))
```

Comment:
This could probably be extracted to a function like:

```cpp
void GameCubeConfigPane::HandleEXISlotChange(int slot, const wxString& title)
{
  assert(slot >= [lower_acceptable_bound] && slot <= [upper_acceptable_bound]);

  if (!m_exi_devices[slot]->GetStringSelection().compare(_(EXIDEV_MIC_STR)))
  {
    InputConfig* const pad_plugin = Pad::GetConfig();
    MicButtonConfigDialog dialog(this, *pad_plugin, title, slot);
    dialog.ShowModal();
  }
  else
  {
    ChooseSlotPath(false, SConfig::GetInstance().m_EXIDevice[slot]);
  }
}
```

and then you just have the event functions forward the necessary information to it.